spark/v2.3/spark/src/main/scala/com/netease/arctic/spark/sql/catalyst/parser/ArcticSqlExtensionsParser.scala

```diff
@@ -0,0 +1,244 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netease.arctic.spark.sql.catalyst.parser
+
+//import com.netease.arctic.spark.sql.parser.{ArcticSqlBaseBaseListener, ArcticSqlBaseLexer, ArcticSqlBaseParser}
+
+import com.netease.arctic.spark.sql.parser.ArcticSparkSqlParser.{NonReservedContext, QuotedIdentifierContext}
+import com.netease.arctic.spark.sql.parser.{ArcticSparkSqlBaseListener, ArcticSparkSqlLexer, ArcticSparkSqlParser}
+import org.antlr.v4.runtime._
+import org.antlr.v4.runtime.atn.PredictionMode
+import org.antlr.v4.runtime.misc.{Interval, ParseCancellationException}
+import org.antlr.v4.runtime.tree.TerminalNodeImpl
+import org.apache.spark.sql.AnalysisException
+import org.apache.spark.sql.catalyst.expressions.Expression
+import org.apache.spark.sql.catalyst.parser.{ArcticExtendSparkSqlAstBuilder, ParseException, ParserInterface}
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+import org.apache.spark.sql.catalyst.trees.Origin
+import org.apache.spark.sql.catalyst.{FunctionIdentifier, TableIdentifier}
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.types.{DataType, StructType}
+
+import java.util.Locale
+
+class ArcticSqlExtensionsParser(delegate: ParserInterface) extends ParserInterface {
+
+  private lazy val createTableAstBuilder = new ArcticExtendSparkSqlAstBuilder(new SQLConf())
+
+  /**
+   * Parse a string to a DataType.
+   */
+  override def parseDataType(sqlText: String): DataType = {
+    delegate.parseDataType(sqlText)
+  }
+
+  /**
+   * Parse a string to a raw DataType without CHAR/VARCHAR replacement.
+   */
+  def parseRawDataType(sqlText: String): DataType = throw new UnsupportedOperationException()
+
+  /**
+   * Parse a string to an Expression.
+   */
+  override def parseExpression(sqlText: String): Expression = {
+    delegate.parseExpression(sqlText)
+  }
+
+  /**
+   * Parse a string to a TableIdentifier.
+   */
+  override def parseTableIdentifier(sqlText: String): TableIdentifier = {
+    delegate.parseTableIdentifier(sqlText)
+  }
+
+  /**
+   * Parse a string to a FunctionIdentifier.
+   */
+  override def parseFunctionIdentifier(sqlText: String): FunctionIdentifier = {
+    delegate.parseFunctionIdentifier(sqlText)
+  }
+
+  /**
+   * Creates StructType for a given SQL string, which is a comma separated list of field
+   * definitions which will preserve the correct Hive metadata.
+   */
+  override def parseTableSchema(sqlText: String): StructType = {
+    delegate.parseTableSchema(sqlText)
+  }
+
+  def isArcticExtendSparkStatement(sqlText: String): Boolean = {
+    val normalized = sqlText.toLowerCase(Locale.ROOT).trim().replaceAll("\\s+", " ")
+     normalized.contains("create table") && normalized.contains("using arctic") && normalized.contains("primary key")
+  }
+
```

Comment:
non primary key table also use arctic grammar. 

arctic grammar support hive partition grammar for unkeyed table .